src/js/browser.js

```diff
@@ -36,68 +36,74 @@ var browser = {
+  var key, tmp;
 
-var detector = {
-    Microsoft_Internet_Explorer: function() { // eslint-disable-line camelcase
-        var detectedVersion = userAgent.match(rIE);
+  var detector = {
+      Microsoft_Internet_Explorer: function() { // eslint-disable-line camelcase
+          var detectedVersion = userAgent.match(rIE);
 
-        if (detectedVersion) { // ie8 ~ ie10
-            browser.msie = true;
-            browser.version = parseFloat(detectedVersion[1]);
-        } else { // no version information
-            browser.others = true;
-        }
-    },
-    Netscape: function() { // eslint-disable-line complexity
-        var detected = false;
+          if (detectedVersion) { // ie8 ~ ie10
+              browser.msie = true;
+              browser.version = parseFloat(detectedVersion[1]);
+          } else { // no version information
+              browser.others = true;
+          }
+      },
+      Netscape: function() { // eslint-disable-line complexity
+          var detected = false;
 
-        if (rIE11.exec(userAgent)) {
-            browser.msie = true;
-            browser.version = 11;
-            detected = true;
-        } else if (rEdge.exec(userAgent)) {
-            browser.edge = true;
-            browser.version = userAgent.match(rEdge)[1];
-            detected = true;
-        } else {
-            for (key in versionRegex) {
-                if (versionRegex.hasOwnProperty(key)) {
-                    tmp = userAgent.match(versionRegex[key]);
-                    if (tmp && tmp.length > 1) { // eslint-disable-line max-depth
-                        browser[key] = detected = true;
-                        browser.version = parseFloat(tmp[1] || 0);
-                        break;
-                    }
-                }
-            }
-        }
-        if (!detected) {
-            browser.others = true;
-        }
-    }
-};
+          if (rIE11.exec(userAgent)) {
+              browser.msie = true;
+              browser.version = 11;
+              detected = true;
+          } else if (rEdge.exec(userAgent)) {
+              browser.edge = true;
+              browser.version = userAgent.match(rEdge)[1];
+              detected = true;
+          } else {
+              for (key in versionRegex) {
+                  if (versionRegex.hasOwnProperty(key)) {
+                      tmp = userAgent.match(versionRegex[key]);
+                      if (tmp && tmp.length > 1) { // eslint-disable-line max-depth
+                          browser[key] = detected = true;
+                          browser.version = parseFloat(tmp[1] || 0);
+                          break;
+                      }
+                  }
+              }
+          }
+          if (!detected) {
+              browser.others = true;
+          }
+      }
+  };
 
-var fn = detector[appName];
+  var fn = detector[appName];
 
-if (fn) {
-    detector[appName]();
+  if (fn) {
+      detector[appName]();
+  }
+}
+else {
+  browser.node = true
```

Comment:
I think this is not enough to detect a node. There is `process` object to detect it. What do you think?

Reply:
That’s fair, we don’t have to assume it’s node and it might be out of scope for what this utility is supposed to do. I can skip that and only check if window and navigator exists, and make no assumptions if it doesn’t.